components/core/src/clp/GlobalMySQLMetadataDB.cpp

```diff
@@ -42,8 +42,9 @@ enum class FilesTableFieldIndexes : uint16_t {
 };
 
 namespace clp {
+
 void GlobalMySQLMetadataDB::ArchiveIterator::get_id(string& id) const {
-    m_db_iterator->get_field_as_string(enum_to_underlying_type(ArchivesTableFieldIndexes::Id), id);
+    m_db_iterator->get_field_as_string(0, id);
```

Comment:
it's a bug fix. the code intends to get the first column of a select statement. It's completely wrong to pass in enum_to_underlying_type(ArchivesTableFieldIndexes::Id) but it happens to work in the past because that enum is 0.

Reply:
I think we should make a constant for it. `clang-tidy` would also complain if we directly use magic numbers like this. (I understand we had similar code in the rest of PR, but I'm fine with the rest for now)

Reply:
Like size_t const first_column_pos = 0?

Reply:
`constexpr` would be more appropriate. Probably make a static const member would be better

Reply:
in my opinion it would be weird to define a static const member just to refer to the first column of any select statement. It makes more sense to define the const in the same function. (so the whole point is to give it a name to not confuse reader with a constant "0" in the code)